plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -333,6 +484,48 @@ private Boolean isModelNotDeployed(MLModelState mlModelState) {
             && !mlModelState.equals(MLModelState.PARTIALLY_DEPLOYED);
     }
 
+    private <T> Boolean isPipelineContainsModel(
```

Comment:
Is there any need for the generic type? Seems that the only return type should be PipelineConfiguration by https://github.com/opensearch-project/OpenSearch/blob/ac443003bad13aeb6373f80439e154dcf25023a2/server/src/main/java/org/opensearch/action/ingest/GetPipelineResponse.java#L86-L88

Reply:
No. ingestion Pipeline configure and search Pipeline configure is different type. https://github.com/opensearch-project/OpenSearch/blob/ac443003bad13aeb6373f80439e154dcf25023a2/server/src/main/java/org/opensearch/action/search/GetSearchPipelineResponse.java#L22

Reply:
I understand there're two classes sharing the same name. Can you take a look at the following example:
```
    // Method that accepts Object and checks the type at runtime
    public void processDate(Object date) {
        if (date instanceof java.util.Date) {
            System.out.println("Processing java.util.Date: " + date);
            // Additional processing for java.util.Date
        } else if (date instanceof java.sql.Date) {
            System.out.println("Processing java.sql.Date: " + date);
            // Additional processing for java.sql.Date
        } else {
            throw new IllegalArgumentException("Unsupported type: " + date.getClass().getName());
        }
    }
```